Kconfig.zephyr

```diff
@@ -442,6 +442,14 @@ config COMPILER_COLOR_DIAGNOSTICS
 	help
 	  Compiler diagnostic messages are colorized.
 
+config COMPILER_STATIC_ANALYSIS
+	bool "Compiler static analysis"
+	depends on "${ZEPHYR_TOOLCHAIN_VARIANT}" = "zephyr"
```

Comment:
Why does this depend on Zephyr SDK?

Reply:
That is GCC specific, while the host toolchain probably will work, that was the closest I got to check that it is using GCC. Is there a better way ?

Reply:
`zephyr` is not the only `ZEPHYR_TOOLCHAIN_VARIANT` that is GCC-based (i.e. there is also the generic `cross-compile`, and crosstool-ng-built `xtools`; though, the latter should/will be removed soon).

The most ideal solution would be to introduce something like `TOOLCHAIN_SUPPORTS_COMPILER_STATIC_ANALYSIS` set by each toolchain variant, and depend on it (similar to how `TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE` works).

But, given that we still have not established a consistent interface for toolchain capability detection (@tejlmand please feel free to correct me if some progress has been made on this front while I was away), I would just leave this an open option with a description saying this option currently only works with GCC (and also potentially a build system warning when an unsupported toolchain is used with this option).

Reply:
I agree.
We should not tie this to Zephyr or gcc toolchains.

Although GCC has a specific `-fanalyze` then other toolchains may supports some form of static analysis during compilation.

That's why we have the compiler properties, which can be left empty if the compiler doesn't have a dedicated flag.

```suggestion
```